### PR TITLE
of biogenerator and biofabricator

### PR DESCRIPTION
**Pull request by ErhardSteinhauer**
## About the PR
Upstream has two machines, capable of making things from biomass - biogenerator and biocube fabricator, but their recipes have overlap and use various amounts of biomass for producing the same stuff (eg, biocube fabricator allows you to print monkey cube for 16 biomass, when biogenerator will use 70 biomass to make the same cube).

This PR makes adjusts material prices for overlapping recipes to be equal, so monkey cubes cost 70 biomass to produce across the board.

Replaced biocube fabricator with biogenerator wherever it was mentioned.

### Biomass requirements adjustments
- Cow cube: 120 -> 350 biomas
- Goat cube: 35 -> 50 biomas
- Mothroach/cockroach/mouse cubes X ->25 biomas, increased crafting times. 
- Biomass requirements for cloning were multiplied by 5.

### New recipes
- "Biowood": 15 biomass -> 1 plank. Available in biogenerator.
- "Biopapersheet": 10 biomass -> 1 paper sheet. Available in biogenerator.
- "Bioplastic": 40 biomass -> 1 plastic sheet. Available in biogenerator.
- Assortment of condiments: 18 biomass per 10u of condiment. Available in biogenerator.
- Raw meat: 45 biomass -> 1 meat. Available in biogenerator.
- Paper sheet to paper pieces: 1 paper piece per 0.1 paper sheet. Available in service techfab.
- Crate with fertile soil (it's a flatpack lol): 1600 for 1 crate of soil (takes 6 min to print). Can be used to construct soil for growing plants. Soil can be deconstructed with shovel. Available in biogenerator. Right now, when you print "crate with fertile soil" and unpack it, you'll get 0 water, 100 nutrient in soil bed, but when you deconstruct a soil bed (need shovel for that btw), you'll get "crate with soil" back, which upon unpacking gives you 0 water and 0 nutrient in soil bed.

## Why / Balance
why have biocube fabricator, when biogenerator exists?

## How to test
Spawn both machines, marvel at biogenerators superiority

## Requirements
- [x] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [x] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes

**Changelog**
:cl: erhardsteinhauer
- tweak: Added more recipes to biogenerator including various animal cubes, bioplastic, biowood, biopapersheet and soil crate.
- tweak: Biogenerator replaced biocube fabricator in research tree.
- tweak: Increased biomass requirements for printing animal/critter cubes and for cloning.
- tweak: Soil beds can now be printed from the biogenerator and assembled with a shovel.

### CODE REVIEW

**Comment by dvir001** [comment]
Is the machine still exists from sci?

**Comment by ErhardSteinhauer** [comment]
> Is the machine still exists from sci?

It exist no more

**Review by whatston3 — COMMENTED** [review]
If mob AI are valuable, wouldn't it be reasonable to increase the mothroach/cockroach/mouse cube cost to some minimum?  Might also be worth a balance pass on the value of produce inserted and the cost of cloning - is an entire person really worth ~10 apples?

Thanks for the pass though, this is useful.

**Comment by ErhardSteinhauer** [comment]
>  is an entire person really worth ~10 apples?

Who can tell?
Biomass, unlike other mats, is integer, so can't go lower than 1. But for sure can bump up cloning biomass numbers.

> If mob AI are valuable, wouldn't it be reasonable to increase the mothroach/cockroach/mouse cube cost to some minimum?

I think the limitation here is the time it takes to print those lil AI shits. Can increase it even further.

File: Content.Server/Cloning/CloningSystem.cs [hunk 1]
```diff
@@ -72,7 +72,8 @@ public sealed class CloningSystem : EntitySystem
         [Dependency] private readonly ISerializationManager _serialization = default!;
 
         public readonly Dictionary<MindComponent, EntityUid> ClonesWaitingForMind = new();
-        public const float EasyModeCloningCost = 0.7f;
+        // Frontier: easy mod is actually hard mode now - 0.7f<5f
+        public const float EasyModeCloningCost = 5f;
```
Comment (whatston3) [inline Content.Server/Cloning/CloningSystem.cs:76]: Would suggest reverting this and setting the cost multiplier (BiomassRequirementMultiplier) in CloningPodComponent.cs to 5.  Not sure this value is configured to get used on the live servers.
```suggestion
        public const float EasyModeCloningCost = 0.7f;
```

**Comment by ErhardSteinhauer** [comment]
> would like see maybe some extra things for future added or talked about like bioplastic , wood , maybe a magnet feature if it doesnt have one yet but thats prob not the right time for it. 300 biomass for cloning (the 5x) is like 55-60 apples then with 50 potency average so it seems better (if cloning was around 60-70 from base)

It already has magnet feature. Adding biowood recipe shouldn't be a problem (I believe, 50 biomass per 1 plank would be fine), but I'm not sure about bioplastic, not because it doesn't make sense, but because it might become extremely easy to print mad amounts of it for botanists if the recipe has reasonable, not entirely unhinged, biomass requirements.

**Comment by ErhardSteinhauer** [comment]
>  i gladly can give some input on how much biomass on average each ship with trays can make.

That would be extremely valuable information to have, yes, please.

**Comment by ErhardSteinhauer** [comment]
Converting to draft until haggling on biomass requirements is concluded.

**Review by whatston3 — COMMENTED** [review]
Pushed a fix for the cloning biomass requirements, had to push the biomassRequirementMultiplier into a DataField.  Printed upgrade speed is correct (not 400% increase in its base state).

One thing I'm not entirely sure about is the ability to mass craft dirt mounds using things you can grow from dirt mounds.  Apart from that, left one minor suggestion on the time to complete material recipes - looks alright to me, happy to approve after minor bits are addressed.

**Comment by ErhardSteinhauer** [comment]
> One thing I'm not entirely sure about is the ability to mass craft dirt mounds using things you can grow from dirt mounds.

Not sure if it is mass craftable with its current price (1200 biomass imo is quite a lot), but can increase both the time (maybe double it, so it'll take 4 minutes to print one or even pump it to 8 mins) it takes to print one and the price (maybe up to 1600).

**Review by whatston3 — APPROVED** [review]

**Comment by whatston3** [comment]
Think both this and #2179 are good to go.  Last call for comment.

**Pull request merged**